sklearn/callback/tests/_utils.py

```diff
@@ -50,10 +50,11 @@ def __init__(self, max_iter=20, computation_intensity=0.001):
         self.computation_intensity = computation_intensity
 
     @_fit_context(prefer_skip_nested_validation=False)
-    def fit(self, X=None, y=None):
-        callback_ctx = self.__skl_init_callback_context__(
-            max_subtasks=self.max_iter
-        ).eval_on_fit_begin(estimator=self)
+    def fit(self, X=None, y=None, X_val=None, y_val=None, callback_ctx=None):
```

Comment:
If we make it public the name should be more explicit, also let's make `callback_context` kwarg only
```suggestion
    def fit(self, X=None, y=None, X_val=None, y_val=None, *, callback_context):
```